File: backend/migration/generate_dummy_reports.py
import mysql.connector
import random

db_config = {
    'user': 'root',
    'password': 'dhanashreessql2025',
    'host': 'localhost',
    'database': 'GMU_Events01'
}

def run():
    try:
        conn = mysql.connector.connect(**db_config)
        cursor = conn.cursor(dictionary=True)

        # 1. Get all completed events
        cursor.execute("SELECT SL_NO FROM event_master WHERE CURRENT_STATUS = 'completed'")
        completed_events = [row['SL_NO'] for row in cursor.fetchall()]

        if not completed_events:
            print("No completed events found.")
            return

        print(f"Found completed events: {completed_events}")

        # 2. Update event_metadata with dummy reports
        dummy_report_text = "This event was executed successfully. All sub-events concluded on time, and participant feedback was overwhelmingly positive. Budget utilization was optimal, and the final outcomes exceeded the initial objectives."
        dummy_pdf_path = "dummy_report.pdf"

        for event_id in completed_events:
            cursor.execute("""
                UPDATE event_metadata 
                SET POST_EVENT_REPORT = %s, REPORT_PDF_PATH = %s 
                WHERE EVENT_ID = %s
            """, (dummy_report_text, dummy_pdf_path, event_id))
            print(f"Updated metadata for event {event_id}")

        # 3. Add dummy registrations and feedback
        dummy_students = ['GMBCAT01', 'GMBCDA01', 'GMBCAI01', 'GMCS01']
        
        comments_list = [
            "Great event, learned a lot!",
            "Well organized and very informative.",
            "The sessions were a bit long but overall good.",
            "Amazing experience, looking forward to the next one.",
            "Good speakers and relevant topics.",
            "Could have had better catering, but the content was top-notch."
        ]

        for event_id in completed_events:
            for student in dummy_students:
                # Check if registration exists
                cursor.execute("SELECT ID FROM event_registrations WHERE EVENT_ID = %s AND STUDENT_ID = %s", (event_id, student))
                reg = cursor.fetchone()

                rating = random.randint(3, 5)
                comment = random.choice(comments_list)

                if reg:
                    # Update existing registration
                    cursor.execute("""
                        UPDATE event_registrations 
                        SET FEEDBACK_RATING = %s, FEEDBACK_COMMENTS = %s, attendance_status = 'Attended', attended = 1, CHECK_IN_STATUS = 'checked_in'
                        WHERE ID = %s
                    """, (rating, comment, reg['ID']))
                else:
                    # Insert new registration
                    cursor.execute("""
                        INSERT INTO event_registrations 
                        (STUDENT_ID, EVENT_ID, STATUS, ROLE, FEEDBACK_RATING, FEEDBACK_COMMENTS, attendance_status, attended, CHECK_IN_STATUS)
                        VALUES (%s, %s, 'active', 'participant', %s, %s, 'Attended', 1, 'checked_in')
                    """, (student, event_id, rating, comment))
            print(f"Added feedback for event {event_id}")

        conn.commit()
        print("Successfully generated dummy data.")

    except Exception as e:
        print(f"Error: {e}")
    finally:
        if 'conn' in locals() and conn.is_connected():
            cursor.close()
            conn.close()

if __name__ == "__main__":
    run()

File: backend/migration/generate_dummy_reports.php
<?php
declare(strict_types=1);

require_once __DIR__ . '/../config/db.php';
$conn = get_db_connection();

// 1. Get all completed events
$result = $conn->query("SELECT SL_NO FROM event_master WHERE CURRENT_STATUS = 'completed'");
$completed_events = [];
while ($row = $result->fetch_assoc()) {
    $completed_events[] = (int)$row['SL_NO'];
}

if (empty($completed_events)) {
    echo "No completed events found.\n";
    exit;
}

echo "Found completed events: " . implode(', ', $completed_events) . "\n";

// 2. Update event_metadata with dummy reports
$dummy_report_text = "This event was executed successfully. All sub-events concluded on time, and participant feedback was overwhelmingly positive. Budget utilization was optimal, and the final outcomes exceeded the initial objectives.";
$dummy_pdf_path = "dummy_report.pdf";

$stmt = $conn->prepare("UPDATE event_metadata SET POST_EVENT_REPORT = ?, REPORT_PDF_PATH = ? WHERE EVENT_ID = ?");
foreach ($completed_events as $event_id) {
    $stmt->bind_param('ssi', $dummy_report_text, $dummy_pdf_path, $event_id);
    $stmt->execute();
    echo "Updated metadata for event $event_id\n";
}
$stmt->close();

// 3. Add dummy registrations and feedback
$dummy_students = ['GMBCAT01', 'GMBCDA01', 'GMBCAI01', 'GMCS01'];
$comments_list = [
    "Great event, learned a lot!",
    "Well organized and very informative.",
    "The sessions were a bit long but overall good.",
    "Amazing experience, looking forward to the next one.",
    "Good speakers and relevant topics.",
    "Could have had better catering, but the content was top-notch."
];

$check_stmt = $conn->prepare("SELECT ID FROM event_registrations WHERE EVENT_ID = ? AND STUDENT_ID = ?");
$update_stmt = $conn->prepare("UPDATE event_registrations SET FEEDBACK_RATING = ?, FEEDBACK_COMMENTS = ?, attendance_status = 'Attended', attended = 1, CHECK_IN_STATUS = 'checked_in' WHERE ID = ?");
$insert_stmt = $conn->prepare("INSERT INTO event_registrations (STUDENT_ID, EVENT_ID, STATUS, ROLE, FEEDBACK_RATING, FEEDBACK_COMMENTS, attendance_status, attended, CHECK_IN_STATUS) VALUES (?, ?, 'active', 'participant', ?, ?, 'Attended', 1, 'checked_in')");

foreach ($completed_events as $event_id) {
    foreach ($dummy_students as $student) {
        $check_stmt->bind_param('is', $event_id, $student);
        $check_stmt->execute();
        $res = $check_stmt->get_result();
        
        $rating = rand(3, 5);
        $comment = $comments_list[array_rand($comments_list)];

        if ($row = $res->fetch_assoc()) {
            $reg_id = $row['ID'];
            $update_stmt->bind_param('isi', $rating, $comment, $reg_id);
            $update_stmt->execute();
        } else {
            $insert_stmt->bind_param('siis', $student, $event_id, $rating, $comment);
            $insert_stmt->execute();
        }
    }
    echo "Added feedback for event $event_id\n";
}

echo "Successfully generated dummy data.\n";
$conn->close();
?>
